x-pack/plugins/encrypted_saved_objects/index.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import crypto from 'crypto';
+import { Request } from 'hapi';
+import { Root } from 'joi';
+import { Legacy } from 'kibana';
+
+import { EncryptedSavedObjectsClientWrapper } from './server/lib/encrypted_saved_objects_client_wrapper';
+import {
+  EncryptedSavedObjectsService,
+  EncryptedSavedObjectTypeRegistration,
+} from './server/lib/encrypted_saved_objects_service';
+import { createLogger } from './server/lib/logger';
+
+export const encryptedSavedObjects = (kibana: any) =>
+  new kibana.Plugin({
+    id: 'encrypted_saved_objects',
+    configPrefix: 'xpack.encrypted_saved_objects',
+    require: ['kibana', 'elasticsearch', 'xpack_main'],
+
+    config(Joi: Root) {
+      return Joi.object({
+        enabled: Joi.boolean().default(true),
+        encryptionKey: Joi.string().min(32),
+        audit: Joi.object({ enabled: Joi.boolean().default(false) }),
+      }).default();
+    },
+
+    async init(server: Legacy.Server) {
+      const config = server.config();
```

Comment:
__question:__ I assume this plugin will be available in Basic so there is no any license related code here?